backend_skeleton/central.py

```diff
@@ -0,0 +1,136 @@
+            data = (await loop.sock_recv(sensor_connect, 1024)).decode('utf-8')
+            print(f"Sensor {sensor_id} : {data}")
+ 
+            # Inform sensor data was received
+            await loop.sock_sendall(sensor_connect,
+                                   (b'You sent: %s' % data.encode('utf-8')))
+            # Append the sensor data
+            sensor_data.append(data)
+            print_data(sensor_data)
+    finally:
+        #Close Connection
+        sensor_connect.close()
+        attached_sensors.remove(sensor_id)
+        print_sensors(attached_sensors)
+
+# Thread for front
+async def serve_front(front_connect, front_address, sensor_data):
+    """Serve the front."""
+    loop = asyncio.get_event_loop()
+    try:
+        print(f"Connected to front! {front_address}")
+        while True:
+            await loop.sock_sendall(front_connect, 
+                                   (bytes('Request Data!',encoding='utf8')))
+            request = (await loop.sock_recv(front_connect, 1024)).decode('utf8')
+            print(f"Request from front : {request}")
+            if request.lower() == 'data':
+              await loop.sock_sendall(front_connect,
+                                (bytes(f'Your Data: {sensor_data}', 
+                                                              encoding='utf8')))
+            elif request.lower() == 'quit':
+                print(f"FRONT has disconnected!")
+                break
+            else:
+                 await loop.sock_sendall(front_connect,
+                                (bytes(f'Unknown Command',encoding='utf8')))
+            # Clear sensor data ( We probably need to save this to database 1st)
+            sensor_data.clear()
+            print_data(sensor_data)
+
+    finally:
+        # Close Connection
+        front_connect.close()
+        # Probably more needs to be done to safely shut down the front if the 
+        # Front disconnects.
+
+
+#Function to print current sensors
+def print_data(sensor_data):
+        """Print connected sensor list."""
+        print("Number of unsent data elements: ", len(sensor_data))
+        print("contents: ", sensor_data)
+
+#Function to print current sensors
+def print_sensors(attached_sensors):
+        """Print connected sensor list."""
+        print("Number of sensors: ", len(attached_sensors))
+        print("List of sensor ID: ", attached_sensors)
+
+#Function to establish a TCP host
+async def server():
+        """Connect to sensors and front"""
+        loop = asyncio.get_event_loop()
+        attached_sensors = []
+        sensor_id = 0
+        sensor_data = []
+        try:
+            tcp_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            server_ip, server_port = 'localhost', 17400
+            host_address = (server_ip, server_port)  # param1 = IP, param2 = port
+            tcp_socket.bind(host_address)
+            tcp_socket.listen(1)
+            tcp_socket.setblocking(False)
```

Comment:
This code seems duplicated in a few place -- perhaps it would be better to create a helper function (possibly in a helper module) to remove the duplication.